website/docs/en/api/javascript-api/compiler.mdx

```diff
@@ -28,10 +28,15 @@ function run(
 ```
 
 :::warning
-This API only supports one compilation at a time. Please call `compiler.close` in the `compiler.run` callback and wait for it to finish before executing `compiler.run` again. Concurrent compilations will damage the output files.
+
+- This API only supports one compilation at a time. Please call `compiler.close` in the `compiler.run` callback and wait for it to finish before executing `compiler.run` again. Concurrent compilations will damage the output files.
+- Direct use of this API requires manual configuration of `compiler.modifiedFiles` and `compiler.removedFiles`, otherwise the memory cache will not be invalidated properly.
+
 :::
 
 ```js
+compiler.modifiedFiles = new Set([...]);
```

Comment:
Letting users set `compiler.modifiedFiles` is not a reasonable API design. We should discuss and provide a more user-friendly API.